modules/sample/src/Module.php

```diff
@@ -25,13 +28,56 @@ public function init()
         parent::init();
         self::$instance = $this;
 
+        $this->_bindEvents();
+
         if (Craft::$app->request->getIsSiteRequest()) {
             Craft::$app->view->registerTwigExtension(new Extension());
         }
 
+        // Always turn on the debug bar in dev environment
+        if (
+            getenv('ENVIRONMENT') === 'dev' &&
+            !Craft::$app->getRequest()->getIsConsoleRequest() &&
+            !Craft::$app->getRequest()->getIsAjax()
+        ) {
+            Craft::$app->session->set('enableDebugToolbarForSite', true);
+        }
+
         Craft::info(
             'Sample module loaded',
             __METHOD__
         );
     }
+
+    private function _bindEvents()
+    {
+        // Add edit entry link to front-end templates
+        Event::on(
+            View::class,
+            View::EVENT_END_BODY,
+            function(Event $e) {
+                $element = Craft::$app->getUrlManager()->getMatchedElement();
+                
+                if (!$element) return;
+
+                $currentUser = Craft::$app->getUser()->identity ?? null;
+
+                if (
+                    Craft::$app->config->general->devMode ||
+                    ($currentUser && $currentUser->can('accessCp'))
+                ) {
+                    $this->view->registerAssetBundle(Bundle::class);
+
+                    echo '<a
+                            href="' . $element->cpEditUrl . '"
+                            class="edit-entry"
+                            target="_blank"
```

Comment:
is this negotiable? guess I can always tweak it ;P

Reply:
Opening in a new window seems like a good default to me. Certainly can modify it to be whatever you want once you've got it setup in your project.

Reply:
What about adding an "external link" arrow to the link for accessibility?

Reply:
Feels a bit like overkill IMO